lib/cosmos/tools/tlm_viewer/widgets/limits_widget.rb

```diff
@@ -34,6 +34,16 @@ def initialize(parent_layout, target_name, packet_name, item_name, value_type, w
     end
 
     def value=(data)
+      if String === data
+        substring = data[0..2]
+        if substring == "Inf".freeze
+          data = Float::INFINITY
+        elsif substring == "-In".freeze
+          data = -Float::INFINITY
+        elsif substring == "NaN".freeze
+          data = Float::NAN
+        end
+      end
```

Comment:
I added this to get consistent behavior for the formatted and with_units cases.  If it looks good please go ahead and merge.